pkg/controller/accountpool/accountpool_controller.go

```diff
@@ -110,13 +110,29 @@ func (r *ReconcileAccountPool) Reconcile(request reconcile.Request) (reconcile.R
 	}
 
 	unclaimedAccountCount := 0
+	claimedAccountCount := 0
 	for _, account := range accountList.Items {
-		if account.Status.Claimed == false && account.Status.State != "Failed" {
-			unclaimedAccountCount++
+		if account.Status.Claimed == false {
+			if account.Status.State != "Failed" {
+				unclaimedAccountCount++
+			}
+		} else {
+			claimedAccountCount++
+		}
+	}
+
+	if currentAccountPool.Status.PoolSize != currentAccountPool.Spec.PoolSize || currentAccountPool.Status.UnclaimedAccounts != unclaimedAccountCount || currentAccountPool.Status.ClaimedAccounts != claimedAccountCount {
+		currentAccountPool.Status.PoolSize = currentAccountPool.Spec.PoolSize
+		currentAccountPool.Status.UnclaimedAccounts = unclaimedAccountCount
+		currentAccountPool.Status.ClaimedAccounts = claimedAccountCount
```

Comment:
Could you make a function here where you pass what you are checking eg. updateStatus(PoolSize) or similar? This line is very long.